src/Testing/BaseMutatorTestCase.php

```diff
@@ -179,19 +180,25 @@ private function getMutationsFromCode(string $code, array $settings): array
 
     private function assertSyntaxIsValid(string $realMutatedCode): void
     {
-        exec(
-            sprintf('echo %s | php -l', escapeshellarg($realMutatedCode)),
-            $output,
-            $returnCode,
-        );
```

Comment:
I think this case pretty much demonstrates why infection itself is so slow when running on real projects - it needs to spawn so many processes.

a few months ago I did a [experiment in phpunit](https://github.com/sebastianbergmann/phpunit/pull/5751#issuecomment-2002582219) to use process forking instead of sub-process creation, which showed a 7-8x performance improvement. maybe thats something which can also be applied to infection.